File: Programmers/모든_문제/연습문제/Solution_Level_1_문자열내마음대로정렬하기.py
def solution(strings, n):
    strings.sort()
    lst = []
    from string import ascii_lowercase
    ls = list(ascii_lowercase)
    for i in ls:
        for j in strings:
            if j[n] == i:
                lst.append(j)         
    return lst
    
def solution1(strings, n):   
    new =[]
    answer =[]
    for i in range(len(strings)):
        a = strings[i][n]
        b = a+strings[i]
        new.append(b)
    new.sort()
    for i in range(len(new)):
        c = new[i][1:]
        answer.append(c)
    return answer
    
def solution2(strings, n): 
    return sorted(sorted(strings), key=lambda x: x[n])

if __name__ == "__main__":
    strings = ["sun", "bed", "car"]
    n = 1
    result = ["car", "bed", "sun"]
    print(result, solution(strings, n), solution1(strings, n), solution2(strings, n))

    strings, n, result = ["ccd", "ccc", "cc", "ccb"], 1, ["ccb", "cc", "ccc", "ccd"]
    print(result, solution(strings, n), solution1(strings, n), solution2(strings, n))
    

File: Programmers/모든_문제/연습문제/Solution_Level_1_문자열내마음대로정렬하기.java
import java.util.Arrays;
import java.util.Comparator;

class Solution_1 {
	
	public static void main(String[] args) {
		// n번째 글자를 기준으로 정렬하기
		System.out.println(Arrays.toString(solution(new String[] {"sun", "bed", "car"},1)));
		System.out.println(Arrays.toString(solution(new String[] {"ccd", "ccc", "cc", "ccb"},1)));
	}
	
	
	public static String[] solution(String[] strings, int n) {
		Arrays.parallelSort(strings, new Comparator<String>() {
			@Override
			public int compare(String o1, String o2) {
				if(o1.charAt(n) == o2.charAt(n)) {
					int idx = 0;
					
					while(idx < o1.length() && idx < o2.length() && o1.charAt(idx) == o2.charAt(idx)) {						
						idx++;
					}					
					
					if(o1.length() == idx) {
						return -1;
					}
					
					if(o2.length() == idx) {
						return 1;
					}
					
					if(o1.charAt(idx) != o2.charAt(idx)) {
						return o1.charAt(idx) - o2.charAt(idx);
					}
					// 완전히 같다면
					return 0;
				}
				return o1.charAt(n) - o2.charAt(n);
			}
		});
		
		return strings;
//		ArrayList<String> li = new ArrayList<>();
//		li.add("ab");
//		li.add("bcd");		
//		return li.toArray(new String[] {});
	}
	
}